src/components/shared/DetailsWidget/detailsWidgetConfig.tsx

```diff
@@ -0,0 +1,221 @@
+import { Address, Colony, ColonyActions, UniversalMessageValues } from '~types';
+import { EventValues, getDetailsForAction } from '~utils/colonyActions';
+import { splitTransactionHash } from '~utils/strings';
+
+import {
+  UserDetail,
+  ActionTypeDetail,
+  RolesDetail,
+  TeamDetail,
+  AmountDetail,
+  DomainDescriptionDetail,
+  ReputationChangeDetail,
+} from '../DetailsWidget';
+
+import styles from './DetailsWidget.css';
+
+const displayName = 'DetailsWidget';
+
+const MSG = defineMessages({
+  activeTeam: {
+    id: `${displayName}.activeTeam`,
+    defaultMessage: 'Active team',
+  },
+  actionType: {
+    id: `${displayName}.actionType`,
+    defaultMessage: 'Action Type',
+  },
+  fromDomain: {
+    id: `${displayName}.fromDomain`,
+    defaultMessage: 'From',
+  },
+  motionDomain: {
+    id: `${displayName}.motionDomain`,
+    defaultMessage: 'Motion created in',
+  },
+  toRecipient: {
+    id: `${displayName}.toRecipient`,
+    defaultMessage: 'To',
+  },
+  value: {
+    id: `${displayName}.value`,
+    defaultMessage: 'Value',
+  },
+  transactionHash: {
+    id: `${displayName}.transactionHash`,
+    defaultMessage: 'Transaction Hash',
+  },
+  domain: {
+    id: `${displayName}.domain`,
+    defaultMessage: 'Team',
+  },
+  domainDescription: {
+    id: `${displayName}.domainDescription`,
+    defaultMessage: 'Team Purpose',
+  },
+  roles: {
+    id: `${displayName}.roles`,
+    defaultMessage: 'Roles',
+  },
+  colonyName: {
+    id: `${displayName}.colonyName`,
+    defaultMessage: 'Name',
+  },
+  reputationChange: {
+    id: `${displayName}.reputationChange`,
+    defaultMessage: `Reputation {isSmiteAction, select,
+        true {penalty}
+        false {reward}
+      }`,
+  },
+  author: {
+    id: `${displayName}.colonyName`,
+    defaultMessage: 'Author',
+  },
+});
+
+const getShortenedHash = (
+  transactionHash: string,
+  showFullDetails: boolean,
+) => {
+  const splitHash = splitTransactionHash(transactionHash);
+  let shortenedHash: string | undefined;
+
+  if (splitHash && !showFullDetails) {
+    const { header, start, end } = splitHash;
+    shortenedHash = `${header}${start}...${end}`;
+  }
+
+  return shortenedHash;
+};
+
+interface DetailItemConfig {
+  label: MessageDescriptor;
+  labelValues?: UniversalMessageValues;
+  item: ReactNode;
+}
+
+const getDetailItems = (
+  actionType: EventValues['actionType'],
+  {
+    motionDomain,
+    fromDomain,
+    toDomain,
+    amount,
+    token,
+    tokenSymbol,
+    reputationChange,
+    isSmiteAction,
+    roles,
+  }: EventValues,
+  colony: Colony,
+  recipientWalletAddress: Address | undefined,
+  transactionHash: string | undefined,
+): DetailItemConfig[] => {
+  const detailsForAction = getDetailsForAction(actionType);
+  const showFullDetails = actionType !== ColonyActions.Generic;
+  const shortenedHash = getShortenedHash(
+    transactionHash || '',
+    showFullDetails,
+  );
+  const colonyName = colony.name;
+
+  return [
```

Comment:
I think this makes it easier to understand what's happening over in `DetailsWidget`, and reduces code duplication.